build-config.json

```diff
@@ -6,10 +6,14 @@
     "srcResources": "./src/content/**/*.@(jpg|jpeg|png|gif)",
     "srcDiagrams": "./src/content/**/*.@(dot|neato|fdp|sfdp|twopi|circo)",
     "srcContentBase": "./src/content",
-    "srcStaticAssets": "./src/@(assets)/**/*.@(jpg|jpeg|png|gif|ttf)",
+    "srcStaticAssets": "./src/@(assets)/**/*.@(jpg|jpeg|png|gif|ttf|js)",
     "srcStyleEntry": "./src/assets/style.scss",
     "srcStylesAny": "./src/assets/**/*.scss",
-    "vendorAssets": ["./node_modules/highlight.js/styles/atom-one-dark.css"],
+    "vendorAssets": [
+      "./node_modules/highlight.js/styles/atom-one-dark.css",
+      "./node_modules/@(htm)/preact/standalone.umd.js",
+      "./node_modules/@(minisearch)/dist/umd/index.js"
```

Comment:
I have added 2 more client side dependencies. Avoiding client bloat is important to me, so I've opted for some lightweight libraries rather than more featureful ones:

* Preact is a lightweight alternative to React. It makes building dynamic frontends a lot easier to reason about and declarative and I would highly recommend using libraries like this over direct DOM manipulation, which is a huge chore. In this case, I am depending also on the companion library "htm" which allows JSX-like syntax without a frontend build step, which I didn't want to get into for this changset. That may be still added in the future if the client features gets more complex. This distribution combines Preact and htm in a single optimized bundle.
* MiniSearch is a small full text search engine implemented in JS. Its search index can be computed offline and loaded from a JSON-serialized format. It's an alternative to something like Lunr.